crates/handlers/src/upstream_oauth2/mod.rs

```diff
@@ -127,7 +128,11 @@ fn client_credentials_for_provider(
             let params = client_secret.ok_or(ProviderCredentialsError::MissingClientSecret)?;
             let params: SignInWithApple = serde_json::from_str(&params)?;
 
-            let key = elliptic_curve::SecretKey::from_pkcs8_pem(&params.private_key)?;
+            let private_key_pem = fs::read_to_string(&params.private_key_file)
```

Comment:
We usually avoid doing sync I/O but rather use Tokio. That means making this function asynchronous and use `tokio::fs::read_to_string` instead